utils/pattern/src/frontend/mod.rs

```diff
@@ -369,56 +367,31 @@ where
                 debug_assert!(actual_len <= max_len);
             }
         }
-        Ok(())
+        if let Some(e) = error {
+            Ok(Err(e))
+        } else {
+            Ok(Ok(()))
+        }
     }
 }
 
-impl<B, P> fmt::Display for WriteablePattern<'_, B, P>
+impl<'b, B, P> fmt::Display for WriteablePattern<'b, B, P>
 where
-    B: PatternBackend,
-    P: PlaceholderValueProvider<B::PlaceholderKey>,
+    B: PatternBackend + 'b,
+    P: PlaceholderValueProvider<B::PlaceholderKey<'b>, Error = B::Error<'b>>,
 {
     #[inline]
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        self.write_to(f)
+        // Discard the TryWriteable error (lossy mode)
+        self.try_write_to(f).map(|_| ())
     }
 }
 
-struct WriteablePatternWithParts<'a, B: PatternBackend, P> {
-    store: &'a B::Store,
-    value_provider: P,
-    literal_part: writeable::Part,
-    element_part: writeable::Part,
-}
-
-impl<B, P> Writeable for WriteablePatternWithParts<'_, B, P>
-where
-    B: PatternBackend,
-    P: PlaceholderValueProvider<B::PlaceholderKey>,
-{
-    fn write_to_parts<S: PartsWrite + ?Sized>(&self, sink: &mut S) -> fmt::Result {
-        for item in B::iter_items(self.store) {
-            match item {
-                PatternItem::Literal(s) => {
-                    sink.with_part(self.literal_part, |w| w.write_str(s))?;
-                }
-                PatternItem::Placeholder(key) => {
-                    let element_writeable = self.value_provider.value_for(key);
-                    sink.with_part(self.element_part, |w| element_writeable.write_to_parts(w))?;
-                }
-            }
-        }
-        Ok(())
-    }
-}
-
-impl<B, P> fmt::Display for WriteablePatternWithParts<'_, B, P>
-where
-    B: PatternBackend,
-    P: PlaceholderValueProvider<B::PlaceholderKey>,
-{
-    #[inline]
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        self.write_to(f)
-    }
+#[test]
+fn test_try_from_store_inference() {
+    use crate::SinglePlaceholder;
+    // Does NOT work:
+    // let _: Pattern<SinglePlaceholder, String> = Pattern::try_from_str("{0} days").unwrap();
+    // Works:
+    let _ = Pattern::<SinglePlaceholder, String>::try_from_str("{0} days").unwrap();
 }
```

Comment:
@Manishearth Rust compiler puzzle. Can you help make the line "Does NOT work" compile like the other line?

Reply:
Yeah this is just a limitation of the resolver.

Reply:
It worked before https://github.com/unicode-org/icu4x/pull/4766/commits/353f3647d305fbaa01d3f4006d808fd6f328c020 though :frowning_face:

Reply:
file upstream imo

Reply:
it's still in the realm of common resolver limitations but the GAT people might like to know

Reply:
This is fixed.